src/codegate/pipeline/codegate_context_retriever/codegate.py

```diff
@@ -149,6 +149,14 @@ async def process(
                         # Combine updated task content with the rest of the message
                         context_msg = updated_task_content + rest_of_message
 
+                elif base_tool == "open interpreter":
```

Comment:
I'm sorry I'm quite dumb today, could you give me a before and after example of the conversations?

Reply:
i actually changed the logic as it was not working in all cases

Reply:
i refactored this code as the logic was not working for all the cases